File: docs/code_examples/cookbook/screenshots_and_pdf/java/playwright_screenshots_pdf.java
import com.microsoft.playwright.*;
import java.nio.file.Paths;

try (Playwright playwright = Playwright.create()) {
    Browser browser = playwright.chromium().connect(
        "wss://playwright.xclouds.dev/v1.58/?api_key=YOUR_API_KEY"
    );
    Page page = browser.newPage();
    page.navigate("https://example.com");

    page.screenshot(new Page.ScreenshotOptions()
        .setPath(Paths.get("screenshot.png")));

    page.pdf(new Page.PdfOptions()
        .setPath(Paths.get("page.pdf")));

    browser.close();
}

File: docs/code_examples/cookbook/screenshots_and_pdf/python/playwright_screenshots_pdf.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.connect(
        ws_endpoint='wss://playwright.xclouds.dev/v1.58/?api_key=YOUR_API_KEY'
    )
    page = browser.new_page()
    page.goto('https://example.com')

    png_bytes = page.screenshot()
    with open('screenshot.png', 'wb') as f:
        f.write(png_bytes)

    pdf_bytes = page.pdf(format='A4')
    with open('page.pdf', 'wb') as f:
        f.write(pdf_bytes)

    browser.close()
